MPI linkage failure

Build failed when `WITH_MPI` enabled:

```
[ 68%] Linking CXX executable chemps2
lto-wrapper: warning: using serial compilation of 26 LTRANS jobs
lto-wrapper: note: see the '-flto' option documentation for more information
/usr/bin/ld: /tmp/ccZKdsWY.ltrans17.ltrans.o: warning: relocation against `ompi_mpi_op_sum' in read-only section `.text'
/usr/bin/ld: /tmp/ccZKdsWY.ltrans0.ltrans.o: in function `file_exists(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)':
<artificial>:(.text+0x721): undefined reference to `ompi_mpi_comm_world'
/usr/bin/ld: <artificial>:(.text+0x73f): undefined reference to `MPI_Comm_rank'
/usr/bin/ld: /tmp/ccZKdsWY.ltrans0.ltrans.o: in function `find_integer(int*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, int, bool, int) [clone .constprop.0]':
<artificial>:(.text+0x92a): undefined reference to `ompi_mpi_comm_world'
/usr/bin/ld: <artificial>:(.text+0x950): undefined reference to `MPI_Comm_rank'
/usr/bin/ld: /tmp/ccZKdsWY.ltrans0.ltrans.o: in function `find_double(double*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, double) [clone .constprop.0]':
<artificial>:(.text+0xc00): undefined reference to `ompi_mpi_comm_world'
/usr/bin/ld: <artificial>:(.text+0xc22): undefined reference to `MPI_Comm_rank'
/usr/bin/ld: /tmp/ccZKdsWY.ltrans0.ltrans.o: in function `find_boolean(bool*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)':
<artificial>:(.text+0xedd): undefined reference to `ompi_mpi_comm_world'

(truncated)
```

(`mpi.h` is installed under `/usr/include`)

This is because [FindMPI](https://cmake.org/cmake/help/latest/module/FindMPI.html) is not performed and MPI libraries are not linked.

--- a/CheMPS2/CMakeLists.txt
+++ b/CheMPS2/CMakeLists.txt
@@ -68,6 +68,9 @@ if (HAS_IPO)
 endif()
 target_include_directories (chemps2-base PRIVATE ${CheMPS2_SOURCE_DIR}/CheMPS2/include/chemps2
                                                  $<TARGET_PROPERTY:tgt::hdf5,INTERFACE_INCLUDE_DIRECTORIES>)
+if (WITH_MPI)
+    target_link_libraries (chemps2-base PUBLIC MPI::MPI_CXX)
+endif ()
 
 if (BUILD_FPIC OR NOT STATIC_ONLY)
     set_target_properties (chemps2-base PROPERTIES POSITION_INDEPENDENT_CODE 1)
@@ -80,7 +83,8 @@ if (NOT STATIC_ONLY)
     endif()
     target_link_libraries (chemps2-shared PRIVATE ${LIBC_INTERJECT})
     target_link_libraries (chemps2-shared PUBLIC  tgt::lapack
-                                                  tgt::hdf5)
+                                                  tgt::hdf5
+                                                  $<$<BOOL:${WITH_MPI}>:MPI::MPI_CXX>)
     set_target_properties (chemps2-shared PROPERTIES SOVERSION ${CheMPS2_LIB_SOVERSION}
                                                      MACOSX_RPATH ON
                                                      OUTPUT_NAME "chemps2"
@@ -94,7 +98,8 @@ if (NOT SHARED_ONLY)
     endif()
     target_link_libraries (chemps2-static PRIVATE ${LIBC_INTERJECT})
     target_link_libraries (chemps2-static PUBLIC  tgt::lapack
-                                                  tgt::hdf5)
+                                                  tgt::hdf5
+                                                  $<$<BOOL:${WITH_MPI}>:MPI::MPI_CXX>)
     set_target_properties (chemps2-static PROPERTIES OUTPUT_NAME "chemps2"
                                                      EXPORT_NAME "chemps2")
 endif()

